lib/proto/proto_conn.c

```diff
@@ -241,9 +242,15 @@ proto_conn_create(int s, struct sock_addr ** sas, int decr, int nopfs,
 		goto err1;
 
 	/* Connect to target. */
-	if ((C->connect_cookie =
-	    network_connect(C->sas, callback_connect_done, C)) == NULL)
-		goto err2;
+	if (sa_b == NULL) {
+		if ((C->connect_cookie =
+		    network_connect(C->sas, callback_connect_done, C)) == NULL)
+			goto err2;
+	} else {
+		if ((C->connect_cookie =
+		    network_connect_bind(C->sas, sa_b, callback_connect_done, C)) == NULL)
```

Comment:
Hmm.  If we changed the network_connect API, these 9 lines could simply be:

```
if ((C->connect_cookie = network_connect_full(C->sas, C->sa_b, NULL, callback_connect_done, C)) == NULL)
    goto err2;
```

@cperciva: the idea is to rename `network_connect_internal` to `network_connect_2` or `network_connect_full` or something like that.  (And expose the renamed `network_connect_internal` in the public API, of course.)

I'm concerned that if we add `network_connect_bind` now, then what happens if we ever want to add `timeo` to that?  Call it `network_connect_bind_timeo`, and then have 4 different functions for opening a network connection?

Reply:
Now that #354 has been merged, this can now be:

```
if ((C->connect_cookie = network_connect_bind(C->sas, C->sa_b, NULL, callback_connect_done, C)) == NULL)
    goto err2;
```

(#355 clarifies that this is safe, but the actual code isn't changing in that PR so it should be fine to change this line now.)

Reply:
Done.